reflex/app.py

```diff
@@ -100,7 +100,47 @@ class OverlayFragment(Fragment):
     pass
 
 
-class App(Base):
+class LifespanMixin:
+    """A Mixin that allow tasks to run during the whole app lifespan."""
+
+    # Lifespan tasks that are planned to run.
+    lifespan_tasks: Set[asyncio.Task] = set()
+
+    @contextlib.asynccontextmanager
+    async def _run_lifespan_tasks(self, app: FastAPI):
+        running_tasks = []
+        try:
+            async with contextlib.AsyncExitStack() as stack:
+                for task in self.lifespan_tasks:
+                    if isinstance(task, asyncio.Task):
+                        running_tasks.append(task)
+                    else:
+                        _t = task()
```

Comment:
should we pass the `app` argument through so the API signature is the same as FastAPI's `lifetime` function